stdlib/public/core/Character.swift

```diff
@@ -125,11 +125,29 @@ public struct Character :
     utf8CodeUnitCount: Builtin.Word,
     isASCII: Builtin.Int1
   ) {
-    self = Character(
-      String(
-        _builtinExtendedGraphemeClusterLiteral: start,
-        utf8CodeUnitCount: utf8CodeUnitCount,
-        isASCII: isASCII))
+    // Most character literals are going to be fewer than eight UTF-8 code
+    // units; for those, build the small character representation directly.
+    if _fastPath(Int(utf8CodeUnitCount) <= 8) {
+      var buffer: UInt64 = ~0
+      _memcpy(
+        dest: UnsafeMutableRawPointer(Builtin.addressof(&buffer)),
+        src: UnsafeMutableRawPointer(start),
+        size: UInt(utf8CodeUnitCount))
+      let utf8Chunk = UInt64(littleEndian: buffer)
+      let bits = MemoryLayout.size(ofValue: utf8Chunk) &* 8 &- 1
```

Comment:
I deliberately kept my logic similar to what was already implemented [elsewhere](https://github.com/apple/swift/blob/6d1ae2a39c1b77240107854b0ae1a35800a8ba73/stdlib/public/core/Character.swift#L180) in the `Character` type. Since I'm not the original author, I can only guess the reason it was written that way—I could imagine resilience being a motivation if the type used for `.small` were to ever change. In that case, I could consider replacing the hardcoded 8 in the main conditional to something `MemoryLayout`-based as well.